### PR TITLE
Fix private property parsing in Flow

### PR DESCRIPTION
**Pull request by kalenikalexander**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | [Parse private class properties with Flow types](https://github.com/babel/babel/issues/8140) 
| Patch: Bug Fix?          |  Yes
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Tests Added + Pass?      | Yes
| Documentation PR Link    | No 
| Any Dependency Changes?  | No
| License                  | MIT

### CODE REVIEW

File: packages/babel-parser/test/fixtures/flow/type-annotations/140/input.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class A {
```
Comment (existentialism) [inline packages/babel-parser/test/fixtures/flow/type-annotations/140/input.js]: I know we do a really bad job of it, but can we give this folder a name? Maybe `class-private-property` or something?

**Comment by existentialism** [comment]
We might also want to make a note to bump https://github.com/babel/babel/blob/master/Makefile#L2 to make sure we're syncing with flow's parsing of private props.

**Comment by kalenikalexander** [comment]
I add `private_class_properties/multiple.js` and `private_class_properties/super.js` to whitelist because parser doesn't do semantic analysis of private properties.

**Comment by radex** [comment]
Hey, what's the status here?  This seems to be the last bit necessary to make private properties work nicely when using Flow.

File: packages/babel-parser/src/types.js [hunk 2]
```diff
@@ -716,6 +716,7 @@ export type ClassPrivateProperty = NodeBase & {
   value: ?Expression, // TODO: Not in spec that this is nullable.
   static: boolean,
   computed: false,
+  typeAnnotation?: ?TypeAnnotationBase, // TODO: Not in spec
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/src/types.js]: Shouldn't this type be ` TypeAnnotation`?

**Comment by existentialism** [comment]
Parsing looks good, should we also add tests to `generator` and `flow-strip-types` to make sure all is well?

**Comment by kalenikalexander** [comment]
I should put `flow-strip-types` tests in `fixtures/regression` or `fixtures/strip-types` directory?

**Comment by nicolo-ribaudo** [comment]
Thank you!

**Pull request merged**